messaging/src/main/java/org/axonframework/eventhandling/DefaultEventProcessorSpanFactory.java

```diff
@@ -0,0 +1,199 @@
+    import java.util.function.Supplier;
+
+    /**
+     * Default implementation of the {@link EventProcessorSpanFactory}.
+     * <p>
+     * Creates a root trace for each batch of events and a linked handler span for each event in the batch by default. This
+     * behavior can be changed by setting {@link Builder#disableBatchTrace(boolean)} to {@code true}.
+     * <p>
+     * If you prefer to have asynchronously handled events in the same trace as the trace that published it, you can set
+     * {@link Builder#distributedInSameTrace(boolean)} to {@code true}. If {@code true}, this duration defaults to 2
+     * minutes, which you can adjust using {@link Builder#distributedInSameTraceTimeLimit(Duration)}.
+     *
+     * @author Mitchell Herrijgers
+     * @since 4.9.0
+     */
+    public class DefaultEventProcessorSpanFactory implements EventProcessorSpanFactory {
+
+        private final SpanFactory spanFactory;
+        private final boolean disableBatchTrace;
+        private final boolean distributedInSameTrace;
+        private final Duration distributedInSameTraceTimeLimit;
+
+        /**
+         * Creates a new {@link DefaultEventProcessorSpanFactory} using the provided {@code builder}.
+         *
+         * @param builder The builder to build the {@link DefaultEventProcessorSpanFactory} from.
+         */
+        protected DefaultEventProcessorSpanFactory(Builder builder) {
+            builder.validate();
+            this.spanFactory = builder.builderSpanFactory;
+            this.disableBatchTrace = builder.builderDisableBatchTrace;
+            this.distributedInSameTrace = builder.builderDistributedInSameTrace;
+            this.distributedInSameTraceTimeLimit = builder.builderDistributedInSameTraceTimeLimit;
+        }
+
+
+        /**
+         * Creates a new {@link Builder} to build a {@link DefaultEventProcessorSpanFactory} with. The default values are:
+         * <ul>
+         *     <li>{@code disableBatchTrace} defaults to {@code false}</li>
+         *     <li>{@code distributedInSameTrace} defaults to {@code false}</li>
+         *     <li>{@code distributedInSameTraceTimeLimit} defaults to 2 minutes</li>
+         * </ul>
+         * The {@code spanFactory} is a required field and should be provided.
+         *
+         * @return The {@link Builder} to build a {@link DefaultEventProcessorSpanFactory} with.
+         */
+        public static Builder builder() {
+            return new Builder();
+        }
+
+        @Override
+        public Span createBatchSpan(boolean streaming, List<? extends EventMessage<?>> eventMessages) {
+            if (distributedInSameTrace || disableBatchTrace || !streaming) {
+                return NoOpSpanFactory.NoOpSpan.INSTANCE;
+            }
+            return spanFactory.createRootTrace(() -> "StreamingEventProcessor.batch");
+        }
+
+        @Override
+        public Span createHandleEventSpan(boolean streaming, EventMessage<?> eventMessage) {
```

Comment:
How come none of these are internal spans as used to be the case with the `AbstractEventProcessor`? Differently put, please explain this decision! :-)

Reply:
While it's very observant, it's different!

Old situation:
- Interceptor in processor implementation created a linked handler trace
- Abstract Event Processor created internal span for processing of messages after interceptorchain was invoked

New situation:
- No interceptor, so no linked handler trace
- AEP creates linked handler trace upon handling in the interceptorchain

Typing this out, there is a step missing. It should be:
- No interceptor, so no linked handler trace
- AEP created linked handler trace around the entire chain
- AEP creates internal trace upon handling in the interceptorchain

This way users can identify how much time is lost in the chain, and, even more important, don't lose the current information.

I will make the adjustments